digitalstrom_mqtt/mqtt.go

```diff
@@ -194,6 +224,254 @@ func (dm *DigitalstromMqtt) getTopic(deviceType string, deviceId string, deviceN
+		return nil, fmt.Errorf("device type not supported %s", device.DeviceType)
+	}
+	device_config := map[string]interface{}{
+		"configuration_url": "https://" + dm.ds_config.Host,
+		"identifiers":       []interface{}{device.Dsid, device.Dsuid},
+		"manufacturer":      "DigitalStrom",
+		"model":             device.HwInfo,
+		"name":              device.Name,
+	}
+	availability := []interface{}{
+		map[string]interface{}{
+			"topic":                 dm.getStatusTopic(),
+			"payload_available":     Online,
+			"payload_not_available": Offline,
+		},
+	}
+	message := map[string]interface{}{}
+	topic := ""
+	if device.DeviceType == digitalstrom.Light {
+		// Setup configuration for a MQTT Cover in Home Assistant:
+		// https://www.home-assistant.io/integrations/light.mqtt/
+		topic = dm.config.HomeAssistantDiscoveryPrefix + "/light/" + device.Dsid + "/light/config"
+		message = map[string]interface{}{
+			"device":            device_config,
+			"name":              utils.RemoveRegexp(device.Name, "light"),
+			"unique_id":         device.Dsid + "_light",
+			"retain":            dm.config.Retain,
+			"availability":      availability,
+			"availability_mode": "all",
+			"command_topic": dm.getTopic(
+				"devices",
+				device.Dsid,
+				device.Name,
+				device.OutputChannels[0],
+				"command"),
+			"state_topic": dm.getTopic(
+				"devices",
+				device.Dsid,
+				device.Name,
+				device.OutputChannels[0],
+				"state"),
+			"payload_on":  "100.00",
+			"payload_off": "0.00",
+			"qos":         0,
+		}
+	} else if device.DeviceType == digitalstrom.Blind {
+		// Setup configuration for a MQTT Cover in Home Assistant:
+		// https://www.home-assistant.io/integrations/cover.mqtt/
+		topic = dm.config.HomeAssistantDiscoveryPrefix + "/cover/" + device.Dsid + "/cover/config"
+		message = map[string]interface{}{
+			"device":            device_config,
+			"name":              utils.RemoveRegexp(device.Name, "cover"),
+			"unique_id":         device.Dsid + "_cover",
+			"device_class":      "blind",
+			"retain":            dm.config.Retain,
+			"availability":      availability,
+			"availability_mode": "all",
+			"state_topic": dm.getTopic(
+				"devices",
+				device.Dsid,
+				device.Name,
+				device.OutputChannels[0],
+				"state"),
+			"state_closed": "0.00",
+			"state_open":   "100.00",
+			"command_topic": dm.getTopic(
+				"devices",
+				device.Dsid,
+				device.Name,
+				device.OutputChannels[0],
+				"command"),
+			"payload_close": "0.00",
+			"payload_open":  "100.00",
+			"payload_stop":  "STOP",
+			"position_topic": dm.getTopic(
+				"devices",
+				device.Dsid,
+				device.Name,
+				device.OutputChannels[0],
+				"state"),
+			"set_position_topic": dm.getTopic(
+				"devices",
+				device.Dsid,
+				device.Name,
+				device.OutputChannels[0],
+				"command"),
+			"position_template": "{{ value | int }}",
+			"qos":               0,
+		}
+		// In case the cover supports tilting.
+		if len(device.OutputChannels) > 1 {
```

Comment:
Are you sure counting the channel is enough for this feature? I guess but isn't there any better way to test this. The channel name has `angle` in it.

Reply:
Here the DigitalStrom API is not very friendly with these OutputChannels. It seems like the second channel is always for the angle (and therefore tilt) so this should work, though I agree is not very robust. Checked for `Angle` in the channel name.

Reply:
Thanks for searching for `Angle`, but could you search for `Angle` in all the output channels (and not just the second one). I have the feeling that this could be an issue if they slightly change their API. 

This is more likely to happen than having multiple "entities" per device in my opinion.

Reply:
Done